dask/utils.py

```diff
@@ -843,6 +843,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     if protocol == 'file':
         # Special case parsing file protocol URL on Windows according to:
         # https://msdn.microsoft.com/en-us/library/jj710207.aspx
+        # urlsplit does not work with ?
+        path = urlpath if "?" in urlpath else path
```

Comment:
What happens if someone provides a URI like `hdfs://user@password/path/to/myfile/data.?.csv`?  I think that we currently use urlsplit for the user/password parsing.